packages/turbo/bin/turbo

```diff
@@ -1,10 +1,257 @@
 #!/usr/bin/env node
 
-const { generateBinPath } = require("../node-platform");
+/**
+ * We need to run a platform-specific `turbo`. The dependency _should_
+ * have already been installed, but it's possible that it has not.
+ */
 
+const child_process = require('child_process');
+const fs = require('fs');
+const path = require('path');
+
+// If we do not find the correct platform binary, should we attempt to install it?
+const SHOULD_INSTALL = true;
+
+// If we do not find the correct platform binary, should we trust calling an emulated variant?
+const SHOULD_ATTEMPT_EMULATED = true;
+
+// Relies on the fact that each tarball publishes the `package.json`.
+// We can simply cd into the `turbo` directory and install there.
+function installUsingNPM() {
+  const turboPath = path.dirname(require.resolve('turbo/package'));
+
+  // Erase "npm_config_global" so that "npm install --global turbo" works.
+  // Otherwise this nested "npm install" will also be global, and the install
+  // will deadlock waiting for the global installation lock.
+  const env = { ...process.env, npm_config_global: undefined };
+
+  child_process.execSync(
+    `npm install --loglevel=error --prefer-offline --no-audit --progress=false`,
+    { cwd: turboPath, stdio: "pipe", env }
+  );
+}
+
+// This provides logging messages as it progresses towards calculating the binary path.
+function getBinaryPath() {
+  // First we see if the user has configured a particular binary path.
+  const TURBO_BINARY_PATH = process.env.TURBO_BINARY_PATH;
+  if (TURBO_BINARY_PATH) {
+    if (!fs.existsSync(TURBO_BINARY_PATH)) {
+      console.warn(
```

Comment:
This is one of the few bits of code that is directly copy pasted. I am also in favor of a hard error here, so I am going to change the behavior. This may break people, but I think we're okay with that.

Reply:
```
Turborepo was unable to find the executable specified by TURBO_BINARY_PATH:
/this/does/not/exist

TURBO_BINARY_PATH is intended for development use-cases. You likely want to unset the environment variable.
```